sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeExtensions.cs

```diff
@@ -99,6 +99,29 @@ internal static FileDownloadInfo ToFileDownloadInfo(this Response<BlobDownloadSt
             };
             return fileDownloadInfo;
         }
+        internal static DataLakeFileReadStreamingResult ToDataLakeFileReadStreamingResult(this Response<BlobDownloadStreamingResult> blobDownloadStreamingResultResponse)
+        {
+            blobDownloadStreamingResultResponse.GetRawResponse().Headers.TryGetValue(Constants.DataLake.EncryptionContextHeaderName, out string encryptionContext);
+            blobDownloadStreamingResultResponse.GetRawResponse().Headers.TryGetValue(Constants.DataLake.AclHeaderName, out string accessControlList);
+            DataLakeFileReadStreamingResult dataLakeFileReadStreamingResult = new DataLakeFileReadStreamingResult()
+            {
+                Content = blobDownloadStreamingResultResponse.Value.Content,
+                Details = blobDownloadStreamingResultResponse.Value.Details.ToFileDownloadDetails(encryptionContext, accessControlList)
+            };
+            return dataLakeFileReadStreamingResult;
+        }
+
+        internal static DataLakeFileReadResult ToDataLakeFileReadResult(this Response<BlobDownloadResult> blobDownloadResult)
+        {
+            blobDownloadResult.GetRawResponse().Headers.TryGetValue(Constants.DataLake.EncryptionContextHeaderName, out string encryptionContext);
+            blobDownloadResult.GetRawResponse().Headers.TryGetValue(Constants.DataLake.AclHeaderName, out string accessControlList);
```

Comment:
Since there's a possibility of this coming back as `null` or default`, should there be a check to check if `TryGetValue` was actually able to get this value?

Because if it is `null` or `default` and gets passed to `ToFileDownloadDetails`, it will definitely throw a null reference exception.

Reply:
I'm glad you brought this up, I also had this thought but dismissed it as "header prob always returns these 2 fields" since the [original convert method](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeExtensions.cs#L89) also does not null check

What are yall's thoughts? @seanmcc-msft @jaschrep-msft

Reply:
Please correct me if I'm wrong here, but upon 2nd look I think that even if the 2 TryGetValue() returned null it should not throw an NRE